content/blog/2023/08/18/2023-08-18-summer-internship-in-jenkins-security.adoc

```diff
@@ -0,0 +1,93 @@
+- wadeck
+opengraph:
+  image: /images/post-images/2023/08/18/2023-08-18-summer-internship-in-jenkins-security.png
+---
+
+=== Context
+
+Jenkins is an open-source CI/CD solution that is extensible with a wide number of plugins that can be installed using the Jenkins plugin distribution repository or via manual installation.
+
+This extensibility is a powerful feature of Jenkins, but on the other hand, it is a critical aspect that has to be secured to avoid risks and vulnerabilities that can impact the Jenkins system.
+
+The internship took place while working in direct contact with the Jenkins Security Team, having daily meetings to discuss topics, possible blockers, and share ideas.
+
+
+=== Internship Goals
+
+The main goals of the internship were:
+
+* Understanding how some of the common vulnerabilities can be recognised in the Jenkins ecosystem by paying attention to common patterns related to best practises in terms of configuration and data flow control;
+
+* Auditing the plugins in a structural and ordered way, with comprehensive vulnerability scoring and classification using the MITRE standard, permitting them to be reviewed easily by the team;
+
+* Reporting findings to the team and plugin maintainers using the Jira issue tracker, having live sessions to help the latter ones have a full comprehension of the vulnerability reported.
+
+
+=== Organisational Model
+
+A good organisational model is based first of all on good communication. Regarding that, a Slack channel was opened where I could ask questions, open discussions, and share ideas.
+
+In addition to that, there were scheduled weekly sessions with my manager, Wadeck Follonier, who leads the Jenkins Security Team, and with my co-pilot, Kevin Guerroudj, where I could open discussion for doubts, receive knowledge, and propose activities that I could do during the internship, as well as a daily session with the team.
+
+To be more productive and get quick feedback from the team, a spreadsheet was made available to me to note the plugins audited with related security audit documents, where I reported my findings and associated security tickets.
+
+Furthermore, CloudBees got me in touch with other interns that worked in different departments using a Slack channel and also scheduled monthly meetings to discuss our progress on our projects and the value added to them.
+
+
+=== The journey
+
+At the beginning of this journey on security, I had the opportunity to quickly learn about the common vulnerabilities that we could find inside a Plugin.
+
+Thanks to a cyber gym that my manager created and distributed on a GitHub repository, it is essentially a Jenkins plugin, called link:https://github.com/Wadeck/emmenthal-plugin[Emmenthal Plugin], that is written with some bad patterns that the same Jenkins documentation suggests to avoid and that permits to obtain common vulnerabilities like Cross Site Scripting, XML External Entities, Server Side Request Forgery, Path Traversal, and so on.
```

Comment:
```suggestion
At the beginning of this security journey, I had the opportunity to quickly learn about the common vulnerabilities that we could find inside a Plugin, thanks to a cyber gym that my manager created and distributed on a GitHub repository.

It is essentially a Jenkins plugin, called link:https://github.com/Wadeck/emmenthal-plugin[Emmenthal Plugin], that is written with some bad patterns that the Jenkins documentation suggests to avoid and that permits to obtain common vulnerabilities like Cross Site Scripting, XML External Entities, Server Side Request Forgery, Path Traversal, and so on.
```
suggestion to rearrange some of the text here